plugins/font/plugin.js

```diff
@@ -209,10 +163,323 @@
+
+		if ( range.collapsed ) {
+			nodeWithStyle = range.startContainer.getAscendant( _hasStyle( configStyleDefinition ), true );
+			return !!( nodeWithStyle && style.checkElementRemovable( nodeWithStyle ) );
+		}
+
+		walker = new CKEDITOR.dom.walker( range );
+		walker.evaluator = CKEDITOR.dom.walker.nodeType( CKEDITOR.NODE_TEXT );
+
+		textNode = walker.next();
+
+		while ( textNode ) {
+			nodeWithStyle = textNode.getAscendant( _hasStyle( configStyleDefinition ) );
+
+			if ( !nodeWithStyle || !style.checkElementMatch( nodeWithStyle ) ) {
+				return false;
+			}
+
+			textNode = walker.next();
+		}
+
+		return true;
+	}
+
+	//  * @param {Object} configStyleDefinition object representing config option:
+	//  * {@link CKEDITOR.config#fontSize_style } or {@link CKEDITOR.config#font_style}
+	//  * @returns {Boolean}
+	function _hasStyle( configStyleDefinition ) {
+		var styleDefinitions = _getAvailableStyleDefinitions( configStyleDefinition );
+
+		return function( el ) {
+			return el.type === CKEDITOR.NODE_ELEMENT && _matchElementToStyleDefinition( el, styleDefinitions );
+		};
+	}
+
+	//  * @param {Object} configStyleDefinition object representing config option:
+	//  * {@link CKEDITOR.config#fontSize_style } or {@link CKEDITOR.config#font_style}
+	//  * @returns {Object[]} Array with objects defining what attributes describe given style e.g.:
+	//  * 	[
+	//  * 		{
+	//  * 			element: 'span',
+	//  * 			attributes: [],
+	//  * 			styles: [ 'font-family' ]
+	//  * 		},
+	//  * 		{
+	//  * 			element: 'font',
+	//  * 			attributes: [ 'face' ],
+	//  * 			styles: []
+	//  * 		}
+	//  * 	]
+	function _getAvailableStyleDefinitions( configStyleDefinition ) {
+		var styleDefinitions = [],
+			objKeys = CKEDITOR.tools.object.keys;
+
+		// Default style types.
+		styleDefinitions.push( {
+			element: configStyleDefinition.element,
+			attributes: objKeys( configStyleDefinition.attributes || {} ),
+			styles: objKeys( configStyleDefinition.styles || {} )
+		} );
+
+		// Style types from override.
+		if ( configStyleDefinition.overrides ) {
+			CKEDITOR.tools.array.forEach( configStyleDefinition.overrides, function( value ) {
+				styleDefinitions.push( {
+					element: value.element,
+					attributes: objKeys( value.attributes || {} ),
+					styles: objKeys( value.styles || {} )
+				} );
+			} );
+		}
+
+		return styleDefinitions;
+	}
+
+	//  * @param {CKEDITOR.dom.element} el
+	//  * @param {Object[]} availableStyleDefinitions - result from `_getAvailableStyleDefinitions()` function
+	function _matchElementToStyleDefinition( el, availableStyleDefinitions ) {
+		for ( var i = 0; i < availableStyleDefinitions.length; i++ ) {
+			var currentStyleDefinition = availableStyleDefinitions[ i ];
+
+			if ( !_hasValidName( el, currentStyleDefinition ) ) {
+				continue;
+			}
+
+			if ( !_hasValidAttributes( el, currentStyleDefinition ) ) {
+				continue;
+			}
+
+			if ( !_hasValidStyles( el, currentStyleDefinition ) ) {
+				continue;
+			}
```

Comment:
**nit:** Maybe marge it into single if statement? Won't be necessary more complicated if separated by a new line:

```javascript
if ( condition1 ||
     condition2 ||
     condition3 ) {
      continue;
}
```